code/ATMOSPHERICS/components/components_base.dm

```diff
@@ -134,7 +134,7 @@ Pipenet stuff; housekeeping
 		for(var/I = 1; I <= device_type; I++)
 			if(nodes["n[I]"])
 				if(parents["p[I]"] == reference)
-					return list("n" = nodes["n[I]"])
```

Comment:
I actually have no idea what brought me to do this
perhaps I thought this fixed the bug that was revealed by fixing this?